internal/server/ptypes/deployment.go

```diff
@@ -0,0 +1,24 @@
+package ptypes
+
+import (
+	"strconv"
+
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+// Type wrapper around the proto type so that we can add some methods.
+type Deployment struct{ *pb.Deployment }
+
+func (v *Deployment) URLFragment() string {
+	// For older deployments (pre WP 0.4.0) we use the sequence. If
+	// we have a generation set, we use the generation initial sequence.
+	seq := v.Sequence
+
+	// By using the generation sequence, we ensure that all deployments
+	// in a generation share the same URL.
+	if g := v.Generation; g != nil {
+		seq = g.InitialSequence
+	}
+
+	return "v" + strconv.FormatUint(seq, 10)
```

Comment:
For mutable deploys with a generation ID, should we append something like `latest` after `v<generation-id>` onto the URL to show they will share the same URL for future deploys?

```
Deployment URL: https://namely-adapting-poodle--v1-latest.waypoint.run
```

Or maybe even just `latest` instead of `v1` if it never changes?

Reply:
I could see where a plugin might do in-place deploys but want to increment the generation id, so maybe something like `v<gen-id>-latest` would work?

Reply:
Generation IDs are long UUIDs though so that'd be a pretty wild URL I think.

Reply:
Ahh, ok. Yeah maybe not then 😅